torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -118,11 +122,19 @@ void GpuLower::lower() {
   }
 
   // Run our passes keeping the lowered expressions and forwarding them
-  const auto lowered_exprs =
-      LoopNestGenerator::loweredExprs(fusion_, fusion_->exprs());
+  const auto lowered_exprs = LoopNestGenerator::loweredExprs(
+      fusion_, reorderExprsForComputeAt(fusion_->exprs()));
```

Comment:
why not inline the reorderExprsForComputeAt() inside lowerExprs()?

Reply:
Agree. That makes more sense to me as `LoopNestGenerator` assumes the incoming expressions are reordered. Updated.

Reply:
Thinking about this again, actually, all the calls with the `Expr*` list assume the incoming list are sorted/transformed in some way. `GpuLower::lower()` is more like a driver of those passes, so having `reorderExprsForComputeAt` directly called from the driver may make more sense, though I don't have strong preference. I'll leave it as is.

Reply:
For exactly the reason you mention, I'd prefer a simpler interface for the "passes" - take in the minimum, strictly required state.

As is, some of the logic leaks out for no obvious reason - calling reorderExprsForComputeAt() is related to what loweredExprs(), not to the driver.

Reply:
Well, that's also the case for pretty much all of the passes. They are not independent and the ordering is important. Having them explicitly called from one place makes the dependency more visible and explicit.

Reply:
are you saying that reorderExprsForComputeAt() is a pass? in that case is should be in the driver, but not as an argument - store the interim IR as a const local variable as the rest of the passes.

Reply:
Agree, we should have such an interface as in standard compilers, not just for lowering but also for optimization.

Reply:
I changed it to look more like a pass.

Reply:
looks much more clear now, thanks. please also add a perf scope for reorderExprsForComputeAt()